### PR TITLE
SDU-1878 fixed npl docu to make it more accurate to Zware API

### PR DESCRIPTION
**Pull request by eternalamenity**

### CODE REVIEW

File: thing-management.md [hunk 1]
```diff
@@ -49,47 +49,33 @@ Object containing nodeId and status.
...
 > `nodeId`: Unique identifier assigned to a device by a Smarthub.
 
 #### `generic_device_class`
 
-Z-Wave generic device class.
-
-Examples:
-
-GENERIC_TYPE_ENTRY_CONTROL, GENERIC_TYPE_SWITCH_BINARY, GENERIC_TYPE_SWITCH_MULTILEVEL.
+Z-Wave generic device class (represented by a number).
```
Comment (BudzikM) [inline thing-management.md:58]: Z-Wave generic device class enum ?

or represented by enum ?

File: thing-management.md [hunk 2]
```diff
@@ -49,47 +49,33 @@ Object containing nodeId and status.
...
-Examples:
-
-SPECIFIC_TYPE_COLOR_TUNABLE_BINARY, SPECIFIC_TYPE_DOOR_LOCK, SPECIFIC_TYPE_CLASS_A_MOTOR_CONTROL
-
-Example of generic and specific device class pair and a matching meaning:
-
-
-GENERIC_TYPE_SWITCH_MULTILEVEL and SPECIFIC_TYPE_CLASS_A_MOTOR_CONTROL means Z-Wave Window Covering device type.
-GENERIC_TYPE_SWITCH_MULTILEVEL and SPECIFIC_TYPE_COLOR_TUNABLE_MULTILEVEL means Z-Wave Color Switch device type.
+Z-Wave specific device class. Along with generic device type creates a specific information about device type (represented by a number).
```
Comment (BudzikM) [inline thing-management.md:62]: who creates 'specific information'?

File: thing-management.md [hunk 3]
```diff
@@ -49,47 +49,33 @@ Object containing nodeId and status.
...
 
 #### `installer_icon_type`
 
-Z-Wave specification has a list of icons with hexadecimal values corresponded to every one of them.
+Z-Wave specification has a list of icons with decimal values corresponded to every one of them.
```
Comment (BudzikM) [inline thing-management.md:66]: corresponding with

File: thing-management.md [hunk 4]
```diff
@@ -49,47 +49,33 @@ Object containing nodeId and status.
...
 
 #### `manufacturer_id`
 
-Unique value given for every manufacturer who sells Z-Wave devices. 
+Unique decimal value given for every manufacturer who sells Z-Wave devices.
```
Comment (BudzikM) [inline thing-management.md:70]: what is decimal value? integer value maybe ?

File: thing-management.md (hunk 1 above)
Comment (eternalamenity) [reply thing-management.md:58]: BY a number.

File: thing-management.md (hunk 2 above)
Comment (eternalamenity) [reply thing-management.md:62]: Specific device class information itself :P

File: thing-management.md (hunk 4 above)
Comment (eternalamenity) [reply thing-management.md:70]: yes

File: thing-management.md (hunk 2 above)
Comment (eternalamenity) [reply thing-management.md]: changed description

File: thing-management.md (hunk 3 above)
Comment (eternalamenity) [reply thing-management.md]: done

File: thing-management.md (hunk 4 above)
Comment (eternalamenity) [reply thing-management.md]: fixed

File: thing-management.md [hunk 5]
```diff
@@ -115,36 +101,46 @@ Described in Node Provisioning Information Type Registry 3.1.2.4.
...
 #### `location`
 
 Location assigned to a device.
+The Location:
+- must be UTF-8 encoded
+- must not contain any appended termination characters
+- may contain the dot
+- must not contain the underscore character “_”.
+- Each location sub-string (separated by the dot character “.”) must not end with the dash character “-”.
+
```
Comment (BudzikM) [inline thing-management.md:130]: can you provide som reges form so someone could take a look how it should be formatted exactly ? liek XX.\d{2} etc

File: thing-management.md [hunk 6]
```diff
@@ -115,36 +101,46 @@ Described in Node Provisioning Information Type Registry 3.1.2.4.
...
+Device location is case sensitive.
+The combined Name and Location strings must not be longer than 62 characters.
 
 #### `joining_info_type`
 
-Number that should be interpreted by a zwave adapter according to table from document Node Provisioning Information Type Registry: 3.1.2.9 Table 5.
+Array of strings which represents joining mode.
+Possible values: ["s0", "s2_unauth", "s2_auth", "s2_ac"]
```
Comment (BudzikM) [inline thing-management.md:137]: Allowed values

File: thing-management.md [hunk 7]
```diff
@@ -225,15 +231,26 @@ Message (command):
...
 
-The following report can occur in five situations:
+Topic: `pt:j1/mt:cmd/rt:ad/rn:zw/ad:1`
 
-1. After adding an entry (status="added").
-2. After deleting an entry (status="deleted" or status="deleted_only_from_npl").
-3. After editing an entry (status="edited", network_status: {"status": "not_in_network"}).
-4. After powering up a device (status="edited", network_status: {"status": "included"}).
-5. After inclusion in wrong network (status="included_bad_network").
+Message (command):
```
Comment (BudzikM) [inline thing-management.md:238]: FIMP message:

File: thing-management.md [hunk 8]
```diff
@@ -115,36 +101,46 @@ Described in Node Provisioning Information Type Registry 3.1.2.4.
 
 Can be one of the following:
 
-- 1: Z-Wave is supported
-- 2: Z-Wave Long Range is supported
+- 0: Z-Wave is supported
```
Comment (BudzikM) [inline thing-management.md:104]: 0:Z-Wave is supported and Long range is not supported. Or too obvious ?

File: thing-management.md (hunk 7 above)
Comment (eternalamenity) [reply thing-management.md:238]: That's how it's done everywhere in FIMP Docu, so You would like to change that, then another PR probably.

File: thing-management.md (hunk 8 above)
Comment (eternalamenity) [reply thing-management.md:104]: It's not said that more values will be available in the future.
I cannot imagine of any now, but it's possible from the frame construction now.

**Pull request merged**